workspaces/core/src/search/private/components/Input.tsx

```diff
@@ -1,95 +1,81 @@
-              </Fragment>
-            );
-          })
-        : null}
+const InnerInput = forwardRef(({
+  placeholder,
+  autoFocus,
+  className
+}: {
+  placeholder?: string;
+  autoFocus?: boolean;
+  className?: string;
+},
+  ref: Ref<HTMLInputElement>) => {
+  const { search, onChangeOpen, onChangeSearch, inputRef } = useContext(SearchContext);
 
-      <div className="flex-1 relative">
-        <input
-          ref={ref}
-          spellCheck={false}
-          className="py-2 px-0 border-0 w-full focus:outline-none focus:border-0 focus:ring-0 bg-transparent placeholder-bulma text-bulma"
-          onChange={(e) => {
-            onChange(e.currentTarget.value);
-          }}
-          onFocus={(e) => {
-            e.currentTarget.select();
-            if (typeof onFocus === 'function') {
-              onFocus();
-            }
-          }}
-          onBlur={onBlur}
-          onKeyDown={(e) => {
-            if (e.key === 'Escape' && value) {
-              e.preventDefault();
-              e.stopPropagation();
-              onChange('');
-            }
-          }}
-          id="moon-search-input"
-          placeholder={placeholder}
-          value={value}
-          type="text"
-          autoFocus={autoFocus}
-        />
+  return <input
+    ref={(ref || inputRef) as LegacyRef<HTMLInputElement>}
+    spellCheck={false}
+    className={
+      mergeClassnames(
+        "py-2 px-0 border-0 w-full focus:outline-none focus:border-0 focus:ring-0 bg-transparent placeholder-bulma text-bulma",
+        className
+      )}
+    onChange={(e) => {
+      onChangeSearch(e.currentTarget.value);
+    }}
+    onFocus={(e) => {
+      e.currentTarget.select();
+      onChangeOpen(true);
+    }}
+    onKeyDown={(e) => {
+      if (e.key === 'Escape' && search) {
+        e.preventDefault();
+        e.stopPropagation();
+        onChangeSearch('');
+      }
+    }}
+    id="moon-search-input"
+    placeholder={placeholder}
+    value={search}
+    type="text"
+    autoFocus={autoFocus}
+  />
+})
 
-        {value && (
-          <button
-            tabIndex={-1}
-            type="button"
-            className="cursor-pointer"
-            onClick={() => {
-              onChange('');
-              const inputElement = document.getElementById('moon-search-input');
-              if (inputElement) {
-                inputElement.focus();
-              }
-            }}
-          >
-            <span className="text-trunks test-sm transition absolute right-4 top-1/2 transform -translate-y-1/2">
-              {clear && clear}
-            </span>
-          </button>
-        )}
-      </div>
-    </div>
-  );
+export const ButtonClear = ({ children }: { children: ReactNode }) => {
+  const { onChangeSearch } = useContext(SearchContext);
+  return <button
+    tabIndex={-1}
+    type="button"
+    className="cursor-pointer text-trunks test-sm transition absolute right-4 top-1/2 transform -translate-y-1/2"
```

Comment:
`test-sm` is this class name correct?  perhaps you mean `text-sm`, if yes please use `text-moon-14`